### PR TITLE
Functional Tests : Migration TS (`@pages\BO` : Part 1)

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Functional Tests : Migration TS (`@pages\BO` : Part 1)
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Relative to #30263 
| Related PRs       | N/A
| How to test?      | Nightly is :green_circle: : https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/3893466101
| Possible impacts? | N/A

### CODE REVIEW

File: tests/UI/campaigns/functional/BO/04_customers/03_outstanding/02_viewOrder.ts [hunk 1]
```diff
@@ -93,11 +93,14 @@ describe('BO - Customers - Outstanding : View order', async () => {
...
+      orderId = parseInt(
+        await ordersPage.getTextColumn(page, 'id_order', 1),
+        10,
+      );
       await expect(orderId).to.be.at.least(1);
 
-      orderReference = await ordersPage.getTextColumn(page, 'reference', 1) as string;
-      await expect(orderReference).to.not.be.null;
+      orderReference = await ordersPage.getTextColumn(page, 'reference', 1);
+      await expect(orderReference).to.not.be.eq('');
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/04_customers/03_outstanding/02_viewOrder.ts]:
```suggestion
      await expect(orderReference).to.not.equal('');
```

File: tests/UI/pages/BO/orders/view/tabListBlock.ts [hunk 2]
```diff
@@ -1,30 +1,129 @@
-require('module-alias/register');
-const ViewOrderBasePage = require('@pages/BO/orders/view/viewOrderBasePage');
+import ViewOrderBasePage from '@pages/BO/orders/view/viewOrderBasePage';
+
+import {Frame, Page} from 'playwright';
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/orders/view/tabListBlock.ts]:
```suggestion
import type {Frame, Page} from 'playwright';
```

File: tests/UI/pages/BO/orders/view/customerBlock.ts [hunk 3]
```diff
@@ -1,15 +1,53 @@
-require('module-alias/register');
-const ViewOrderBasePage = require('@pages/BO/orders/view/viewOrderBasePage');
+import addAddressPage from '@pages/BO/customers/addresses/add';
+import ViewOrderBasePage from '@pages/BO/orders/view/viewOrderBasePage';
 
-// Needed to create customer in orders page
-const addAddressPage = require('@pages/BO/customers/addresses/add');
+import type {Page} from 'playwright';
+import {Frame} from 'playwright';
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/orders/view/customerBlock.ts]:
```suggestion
import type {Frame, Page} from 'playwright';
```

File: tests/UI/pages/BO/orders/view/viewOrderBasePage.ts [hunk 4]
```diff
@@ -186,9 +235,9 @@ class ViewOrderBasePage extends BOBasePage {
...
    * @returns {Promise<boolean>}
    */
-  isReturnProductsButtonVisible(page) {
+  isReturnProductsButtonVisible(page: Page): Promise<boolean> {
     return this.elementVisible(page, this.returnProductsButton, 2000);
   }
 }
 
-module.exports = new ViewOrderBasePage();
+module.exports = ViewOrderBasePage;
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/orders/view/viewOrderBasePage.ts]:
```suggestion
```

File: tests/UI/pages/BO/orders/view/viewOrderPage.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import ViewOrderBasePage from '@pages/BO/orders/view/viewOrderBasePage';
+
+/**
+ * @class
+ * @extends ViewOrderBasePage
+ */
+class ViewOrderPage extends ViewOrderBasePage {
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/orders/view/viewOrderPage.ts]: Maybe something to improve later : 
```js
// viewOrderBasePage.ts
const viewOrderBasePage = new ViewOrderBasePage();
export {viewOrderBasePage, ViewOrderBasePage};
```

1. Import `ViewOrderBasePage` in pages to extend it
2. Import `viewOrderBasePage` in test to use it

So you can delete this file

File: tests/UI/pages/commonPage.ts [hunk 6]
```diff
@@ -5,7 +5,7 @@ import type {
 } from '@data/types/playwright';
 
 import {
```
Comment (boubkerbribri) [inline tests/UI/pages/commonPage.ts]:
```suggestion
import type {
```

**Review by boubkerbribri — COMMENTED** [review]
some little suggestions

File: tests/UI/campaigns/functional/BO/04_customers/03_outstanding/02_viewOrder.ts (hunk 1 above)
Comment (Progi1984) [reply tests/UI/campaigns/functional/BO/04_customers/03_outstanding/02_viewOrder.ts]: Fxied

File: tests/UI/pages/BO/orders/view/customerBlock.ts (hunk 3 above)
Comment (Progi1984) [reply tests/UI/pages/BO/orders/view/customerBlock.ts]: Fixed

File: tests/UI/pages/commonPage.ts (hunk 6 above)
Comment (Progi1984) [reply tests/UI/pages/commonPage.ts]: Fixed

File: tests/UI/pages/BO/orders/view/viewOrderPage.ts (hunk 5 above)
Comment (Progi1984) [reply tests/UI/pages/BO/orders/view/viewOrderPage.ts]: Fixed

File: tests/UI/pages/BO/orders/view/viewOrderBasePage.ts (hunk 4 above)
Comment (Progi1984) [reply tests/UI/pages/BO/orders/view/viewOrderBasePage.ts]: Fixed

File: tests/UI/pages/BO/orders/view/viewOrderBasePage.ts [hunk 7]
```diff
@@ -1,12 +1,61 @@
...
+
+import type {Page} from 'playwright';
 
 /**
  * View order base page, contains functions that can be used on view/edit order page
  * @class
  * @extends BOBasePage
  */
-class ViewOrderBasePage extends BOBasePage {
+export default class ViewOrderBasePage extends BOBasePage {
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/orders/view/viewOrderBasePage.ts]:
```suggestion
class ViewOrderBasePage extends BOBasePage {
```

to not export it 2 times maybe

Comment (Progi1984) [reply tests/UI/pages/BO/orders/view/viewOrderBasePage.ts]: Fixed

**Review by NeOMakinG — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by boubkerbribri** [comment]
Thanks @Progi1984

**Pull request merged**